### DISCUSSION TITLE
How to use config.resolve.alias with storybook/nextjs settings

### DISCUSSION THREAD

**Discussion by yama-sitter** [storybookjs/storybook#25785] [Help]
### Summary

Hello. 

When we upgraded Nextjs to v13, we also introduced **storybook/nextjs**.

We are originally using Storybook to run the Visual Regression Test, and we are replacing files using the `config.resolve.alias` property to avoid differences in the Visual Regression Test.

However, it seems that the **tsconfig-paths-webpack-plugin** setting used internally by **storybook/nextjs** takes precedence, so the setting of the `paths` property in tsconfig.json takes effect instead of the existing `config.resolve.alias` setting.

Is this a storybook/nextjs specification?
Is there a workaround?

The following is a description of the `webpackFinal` property settings.
This is not the actual code, but the configuration is the same as the actual one.

```
const config = {
...
  async webpackFinal(config) {
    ...
    config.resolve.alias = {
      ...config.resolve.alias,
      '~/xxModule': resolve(__dirname, '../src/sb-build/xxModule/'),
      '~': resolve(__dirname, '../src/'),
    };

    return config;
  },
  ...
}
```

**Comment by valentinpalkovic** [comment]
Couldn't you just go to your `.storybook` folder and add the `paths` property to its `tsconfig.json`? If a tsconfig.json doesn't exist, just create one, extend your main tsconfig and add the necessary `paths` configuration.

**Reply by yama-sitter** [reply]
Thanks for the answer!

However, I think the paths property in tsconfig.json does not allow "file path resolution".
When I run Storybook, I want to replace "specific files" instead of "specific directories".
Can this be solved by the method you have presented?

**Reply by valentinpalkovic** [reply]
Could you try it out to map single files via paths mapping?
